### DISCUSSION TITLE
How to set breakpoint as an init-command

### DISCUSSION THREAD

**Discussion by mortymacs** [go-delve/delve#3189] [Q&A] [answered]
Hi,

In GDB, there is a flag `--init-command` which I can define a command like `b dbConnection` and it will be applied automatically after running the project. It sets the breakpoint on the `dbConnection` function.

I was looking for the same in Delve but I couldn't find it.
The whole idea is instead of running dlv command and then telling where is my function and set the breakpoint there, I wanna give the function name before running dlv as init command like `break pkg/db/connection.go:dbConnection`. I can run fzf command to find the file easily and then I'll pass it to the dlv init command to set the breakpoint there.

Cheers,
Mort

**Comment by aarzilli** [comment, marked as answer]
You can use the init option, for example `dlv debug --init <(echo b main.main)`

**Reply by mortymacs** [reply]
Thank you so much!
